contracts/base/OnlySelfCallable.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.18;
+
+abstract contract OnlySelfCallable {
+    error InvalidSender(address sender);
+    error InvalidCalldataLength();
+
+    modifier onlySelf() {
```

Comment:
Similar to my comment below (which was resolved) this function doesn't make sense for me. Same the contract name. "Self" here is the manager contract, which is not relevant in any check. We are checking that the call comes from the account ... so something like "onlyAccount" or similar would make more sense